rfcs/inlang-git-server-tech-stack/RFC.md

```diff
@@ -0,0 +1,80 @@
+---
+# the frontmatter is only relevant for rendering this site on the website
+title: inlang git server tech stack
+description: The tech stack for the inlang git server
+---
+
+# inlang git server tech stack
+
+## Overview
+
+We will be building a (variant) of git-server soon as part of [the next git vision 2023](https://gist.github.com/araknast/d80f8b64b2777e42b49fd9e89e5a1b13).
+
+## Requirements
+
+- TypeScript to increase iteration speed (no Go, Rust, etc.) [High Confidence]
```

Comment:
What do we deem of more importance, iteration speed or performance? 

I am in favor of iteration speed. If the client is written in TS, writing the server in TS will be faster. If the server should be a superset of the client (https://github.com/inlang/inlang/issues/822), TS is a must. 

We can improve performance at a later stage by a) incrementally introducing WebAssembly for performance-critical code and/or b) re-write the stack in a lower-level language in the future. Although, b) would lead to a) if the stack should be run in JavaScript/in the browser.  Eh, given that b) leads to a), the choice of TS for the server seems like a no-brainer.

Reply:
I think performance is critical, since a git server must do a significant amount of work in unpacking and repacking packfiles to send to the client. However, this is a fairly static area, so writing only this portion of the code in WASM could be a solution. In fact, perhaps we could experiment with integrating WASM into isomorphic-git in order to speed up performance pain points such as checkout and packfile delta resolution.

As far as a p2p git network goes, I am still working out the implementation details, bit I think it would make sense to have this as a separate codebase from the client-server model server code. In that case, I think TS with performance critical elements written in WASM makes the most sense here.